### ISSUE TITLE
Please provide information on how to use the plugin triggers from a pipeline script

### ISSUE THREAD

**Issue by gortvab** [jenkinsci/jetbrains-space-plugin#17]
### Describe your use-case which is not covered by existing documentation.

Hi,

My company is trying to set up the integration between Space and Jenkins with this plugin.

So far we have set up the Jenkins application in Space and we have managed to connect the two together.
We have set up the webhook to connect to Jenkins when a branch event happens.
In the webhook log, we can see that the POST request itself is sent to Jenkins, however, we get a `400 Bad Request` response back from Jenkins and our jobs don't get triggered.

As we see, Jenkins can't seem to find the job based on the webhook id.

Here's an extract of our Jenkins log:
```
Mar 28, 2024 4:55:13 PM WARNING org.jetbrains.space.jenkins.trigger.SpaceWebhookEndpointKt processWebhookCallback
Trigger found for webhook id = 3G5FnG3hMRi3 found by fallback to webhook name

Mar 28, 2024 4:55:13 PM WARNING org.jetbrains.space.jenkins.trigger.SpaceWebhookEndpointKt processWebhookCallback
No registered trigger found for webhook id = 3G5FnG3hMRi3
```

Could you provide some information in the documentation on how to properly set the triggers up in Jenkins?

We would also like to use this plugin from a pipeline script, however we are unsure how to set up the pipeline trigger for Space events.

Here's what we have tried:
```groovy
stage("TEST") {
    steps {
        script {
            properties([pipelineTriggers([
                [
                    $class: 'SpaceWebhookTrigger',
                    allowSafeMergeWithBranches: false,
                    allowSafeMergeWithMergeRequests: false,
                    id: 'secret',
                    projectKey: 'project',
                    repositoryName: 'repo-name',
                    spaceConnectionId: 'connection'
                ]
            ])
            ])
        }
    }
}
```
Would the trigger work like this, or should we try another way?
Based on the current documentation, it is unclear which events could be used from a pipeline script.

Any help would be much appreciated.

Thank you in advance :)

Best Regards,
Boglárka

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

We previously used the Bitbucket Push and Pull request plugin for triggering builds. This plugin documentation might be useful for inspo:
https://plugins.jenkins.io/bitbucket-push-and-pull-request/#plugin-content-pipeline-script

**Comment by AndySpu** [comment]
Hi @gortvab 

I'm in the process of integrating Space and Jenkins into my enterprise.

I solved the Jenkins trigger from Space via the "Generic Webhook Trigger" Plugin for Jenkins.

In Jenkins pipeline I set up within Build Triggers "Generic Webhook Trigger" and added a token.
![image](https://github.com/jenkinsci/jetbrains-space-plugin/assets/19991266/aafebc77-79a9-42d0-813a-246388a9af05)


In Space it is required to Create new Webhook in the Application and subscribe this webhook to an event within Space. Also it is required to set Endpoint of the Webhook in Jenkins.
![image](https://github.com/jenkinsci/jetbrains-space-plugin/assets/19991266/28c23d9c-f084-4213-8ea2-60685935811b)

**Comment by Kiryushin-Andrey** [comment]
Hi @gortvab @AndySpu

The plugin provides a way to set up build triggers without dealing with the Space webhooks manually, described here - https://github.com/jenkinsci/jetbrains-space-plugin?tab=readme-ov-file#triggering-builds. It looks like this in the pipeline settings:

<img width="1207" alt="image" src="https://github.com/jenkinsci/jetbrains-space-plugin/assets/11661335/6caffbe5-4a2b-4151-810a-3b2fccfea3d8">

Under the hood, it does the same as @AndySpu did manually - it installs a webhook for the relevant events in Space, pointing to an endpoint provided by the plugin. Have you tries this? Please let me know if there are any problems with this approach.

**Comment by AndySpu** [comment]
Hi @Kiryushin-Andrey 
thank you for rapid reply.

I did try to use the trigger from the plugin. When saving the Jenkins Job configuration, I even found the webhook that Jenkins created in Space. But when trying to test the connection, Jenkins responds with a status of 400

![image](https://github.com/jenkinsci/jetbrains-space-plugin/assets/19991266/681810f7-389e-4163-888d-1c44d0e31111)

Tested on Jenkins 2.440.1 and Space 2024.01 (on-premises)

**Comment by Kiryushin-Andrey** [comment]
Ah, I see now. I believe you might have the same issue as reported here - https://github.com/jenkinsci/jetbrains-space-plugin/issues/18, I'll look into it shortly. Meanwhile, there is even a workaround suggested there, but it requires tweaking plugin source code. Setting up the webhook manually as you did is another valid workaround for now

**Comment by AndySpu** [comment]
By using the generic webhook work around i am running into another issue.
The typical scenario in jenkins-space integration is the case, where i need to build particular branch for what merge request was created. In the pipeline configuration i am using JetBrains Space as SCM with following configuration:

![image](https://github.com/jenkinsci/jetbrains-space-plugin/assets/19991266/6f1fc341-152c-48d4-b4b1-ee0fc2033898)
![image](https://github.com/jenkinsci/jetbrains-space-plugin/assets/19991266/a39138ef-a7c4-4fd8-8362-eecfa96d31a3)


every time a job is performed, the checkout is against the master. Is there a way to change it?

**Comment by Kiryushin-Andrey** [comment]
I've fixed the initial issue #18, so you can update to the latest release and set up the plugin-provided build triggers, they should work now.

As for you last question, I've digged into it a bit and unfortunately it is not currently possible at the moment to include the name of a merge request source branch into the custom payload of a Space webhook about MR creation or update. If it was possible, then Jenkins job could be parameterized with branch name, Space webhook could be set up to invoke `buildWithParameters` Jenkins endpoint passing in the branch name, and this branch name parameter could then be used  in the "Branch specifier" field of the git checkout settings to check out the right branch.
I've filed an issue for this - https://youtrack.jetbrains.com/issue/SPACE-20274, you can subscribe to it if you're interested.
